client/src/testing/activated-route-stub.ts

```diff
@@ -19,8 +19,10 @@ export class ActivatedRouteStub {
   /** The mock paramMap observable */
   readonly paramMap = this.subject.asObservable();
 
-  /** Set the paramMap observables's next value */
+  /** Set the paramMap to the next value of the observable */
   setParamMap(params?: Params) {
```

Comment:
I wonder why this had an optional param in the first place, it seems like this could be simplified by making it a required param and removing the check

Reply:
It's a slightly odd design. The constructor takes that as an optional argument, and then calls `setParamMap()` with that (optional) value. I presume that's why it's marked as optional, because it inherits that from the constructor. The way things are _used_ is that the constructor _never_ gets an argument, but `setParamMap` _always_ gets one.

A plausible option would be to make the `setParamMap` argument required, and have the constructor check to see if it was actually given a value and only call `setParamMap` if it was.

Reply:
I think this relates to discussion in #53